tests/unit/misc/test_lineparser.py

```diff
@@ -53,24 +54,89 @@ def test_prepare_save_missing(self, mocker, lineparser):
         lineparser._prepare_save()
         os_mock.makedirs.assert_called_with(self.CONFDIR, 0o755)
 
+    def test_prepare_save_no_config(self, mocker):
+        """Test if _prepare_save doesn't create a None config dir."""
+        os_mock = mocker.patch('qutebrowser.misc.lineparser.os')
+        os_mock.path.exists.return_value = True
+
+        lineparser = lineparsermod.BaseLineParser(None, self.FILENAME)
+        assert not lineparser._prepare_save()
+        assert not os_mock.makedirs.called
+
+    def test_double_open(self, mocker, lineparser):
+        """Test if _open refuses reentry."""
+        mocker.patch('builtins.open', mock.mock_open())
+
+        with lineparser._open('r'):
+            with pytest.raises(IOError):
+                with lineparser._open('r'):
+                    pass
+
+    def test_binary(self, mocker):
+        """Test if _open and _write correctly handle binary files."""
+        open_mock = mock.mock_open()
+        mocker.patch('builtins.open', open_mock)
+
+        testdata = b'\xf0\xff'
+
+        lineparser = lineparsermod.BaseLineParser(
+            self.CONFDIR, self.FILENAME, binary=True)
+        with lineparser._open('r') as f:
+            lineparser._write(f, [testdata])
+
+        open_mock.assert_called_once_with(
+            os.path.join(self.CONFDIR, self.FILENAME), 'rb')
+
+        open_mock().write.assert_has_calls([
+            mock.call(testdata),
+            mock.call(b'\n')
+        ])
+
 
 class TestLineParser:
 
+    """Tests for LineParser."""
+
```

Comment:
FWIW I got rid of (some of) those comments as they just say the same thing the class name already says, so I'd prefer this to not be added again.